### ISSUE TITLE
pytorch 2.13.0+gfx906: non-deterministic GPU crash under large-sequence attention (Qwen-Image-Edit, ~12k tokens) — HSA_STATUS_ERROR_INVALID_PACKET_FORMAT + gfx_v9_0_bad_op_irq

### ISSUE THREAD

**Issue by trangsinhdia** [mixa3607/ML-gfx906#27]
**Hardware:** MI50 32GB (gfx906, `sramecc-:xnack-`), Ryzen 7 9700X, host kernel 7.0.0-30-generic.
**Image:** `mixa3607/comfyui-gfx906:latest-rocm-7.14` (ComfyUI 0.33.1, torch `2.13.0+gfx906.20260802001858`, ROCm 7.14).

**Workload:** ComfyUI, Qwen-Image-Edit-2511 Q8_0 GGUF (unsloth) with 2 reference images -> ~12.3k-token self-attention, ComfyUI sub-quadratic attention path (default on gfx906), `bfloat16 weights, manual cast fp32`.

Crash is **non-deterministic** (~3/14 runs at 20-40 steps; shorter runs crash too). Two flavors observed:

1. Mid-sampling (ROCm 7.14):
```
:0:rocdevice.cpp :4212: Callback: Queue 0x... aborting with error :
HSA_STATUS_ERROR_INVALID_PACKET_FORMAT: The AQL packet is malformed.
```
dmesg: `[drm:gfx_v9_0_bad_op_irq [amdgpu]] *ERROR* Illegal opcode in command stream`

2. During partial-load weight streaming (`torch.nn.Module.to()` H2D copy, right after sampling finished):
`amdgpu: [mmhub0] no-retry page fault ... from IH client 0x12 (VMC)` at a host VA -> SIGSEGV/Abort.

Same workload on your rocm-6.3.3 image (torch 2.11) also crashes (GPU page fault at step 6/20) — so not a 7.14 regression. Text-to-image (~4k tokens) and 1-ref edit (~8k tokens): 0 crashes ever. llama.cpp (`b10327-rocm-7.14`) on the same card runs 30+ GiB models daily without issues.

**Questions:**
1. Release 20260808180342 applied `-mllvm -amdgpu-sched-strategy=max-ilp` to fix gfx906 codegen **for llama.cpp only** — should pytorch be rebuilt with the same scheduler fix? The illegal-opcode signature suggests the compiler emits an instruction gfx906 doesn't implement, on some large-GEMM kernel variants.
2. Is hipBLASLt fallback expected to be fully disabled on gfx906 builds? (We set `ROCBLAS_USE_HIPBLASLT=0` defensively.)

Happy to run diagnostic builds / `ROCBLAS_LAYER=2` traces on demand.
Cross-ref: ROCm/TheRock#1844 (gfx906 unvalidated in ROCm 7.x, stream-sync primitive test failures confirmed by AMD).

**Comment by mixa3607** [comment]
Sorry, I don't know when I'll be able to debug this issue
